java-lib/src/main/java/com/yammer/metrics/core/DeltaCounter.java

```diff
@@ -0,0 +1,40 @@
+package com.yammer.metrics.core;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.wavefront.common.MetricConstants;
+
+/**
+ * A counter for Wavefront delta metrics.
+ *
+ * Differs from a counter in that it is reset in the SocketMetricsProcessor every time the value is
+ * reported.
+ *
+ * (This is similar to how WavefrontHistogram is implemented)
+ *
+ * @author Suranjan Pramanik (spramanik@wavefront.com)
+ */
+public class DeltaCounter extends Counter {
+
+    private DeltaCounter() {
+        // nothing to do. keeping it private so that it can't be instantiated directly
+    }
+
+    @VisibleForTesting
+    public static synchronized DeltaCounter get(MetricsRegistry registry, MetricName metricName) {
+        if (registry == null || metricName == null || metricName.getName().isEmpty()) {
+            throw new IllegalArgumentException("Invalid arguments");
+        }
+
+        String name = metricName.getName();
+        if (!(name.startsWith(MetricConstants.DELTA_PREFIX) || name.startsWith(MetricConstants
+                .DELTA_PREFIX_2))) {
+            name = MetricConstants.DELTA_PREFIX + name;
+        }
+
+        DeltaCounter counter = new DeltaCounter();
+        MetricName newMetricName =  new MetricName(metricName.getGroup(), metricName.getType(),
```

Comment:
Are we sure we should use this constructor? We should look at equals() operation and make sure it works for DeltaCounters? add a test for it if you haven't already.

Reply:
good catch. Used a different constructor. This required some more changes in the code.